core/services/ping/ping1d_mavlink.py

```diff
@@ -55,7 +55,7 @@ def distance_message(time_boot_ms: int, distance_cm: int, device_id: int, confid
             "horizontal_fov": 0.52,
             "vertical_fov": 0.52,
             "quaternion": [0, 0, 0, 0],
-            "signal_quality": confidence,
+            "signal_quality": min(1, confidence),  # 0 means undefined per MAVLink spec
```

Comment:
0 means out of reach or nothing detect IIRC. But in either case, why are you removing 0 ?

Reply:
Ok, I forgot to check the issue, let us trust the user.